pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -876,8 +868,28 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 						Service:                    service,
 					}
 
-					configgen.buildSidecarOutboundListenerForPortOrUDS(node, listenerOpts, pluginParams, listenerMap,
-						virtualServices, actualWildcard)
+					// Minor optimization for Kubernetes statefulsets/headless services
+					// Instead of generating a single 0.0.0.0:Port listener, generate a listener
+					// for each podIP.
+					if bind == "" && service.Resolution == model.Passthrough &&
+						service.Attributes.ServiceRegistry == string(serviceregistry.KubernetesRegistry) {
```

Comment:
Why? What about multicluster? Or if we serve kubernetes config over mcp

Reply:
It does not relate to multicluster here.

Reply:
I dont think MCP matters here because we can ensure that the attribute is always kubernetes by encoding the source of the data in the service entry annotations. We need it after all for other areas.